### ISSUE TITLE
Update benchmarks

### ISSUE THREAD

**Issue by michaelkirk** [georust/geographiclib-rs#19]
From https://github.com/georust/geographiclib-rs/pull/17#issuecomment-763277044

> I would favor also expanding the benchmarks before starting on troubleshooting the core code. This might boil down to offering the same builtin/small/full options on the rs benches that you added to the unit tests. I suggest replacing the benches' current data-copying approach with Arc+Mutex at the same time. That improved speed by about 66% in some benches I compared, though it may depend in part on the size of the test dataset.

**Comment by stonylohr** [comment]
@michaelkirk I'll prepare a PR for this, assuming that works for you.

**Comment by michaelkirk** [comment]
Please do!

**Comment by stonylohr** [comment]
I withdraw my suggestion of using Arc and Mutex instead of cloning.

It looks like using Arc and Mutex instead of cloning inputs is a little more expensive for the builtin and short modes, and a little less expensive for the full mode. All the differences may just be noise, nothing dramatic like I saw in my earlier scenario. Thinking back on it, my inputs in the earlier scenario may have been a vector of vectors which is probably much more expensive to clone than a vector of tuples. I'll submit a PR with the remaining changes shortly.

Note that I also haven't looked into integrating it with bors yet.

It may be a couple of days until I can get back to this. You're welcome to do the bors integration yourself in the meantime, or wait til I can get back to it.

**Comment by stonylohr** [comment]
Actually, I used the existing CI PR, since that branch is still the only one with all the other CI-related changes. I can reformulate if you'd like it in a separate PR.

**Comment by stonylohr** [comment]
@michaelkirk I think we can close this. Do you agree?

**Comment by michaelkirk** [comment]
Agree, thanks!
